src/Glpi/Http/Kernel.php

```diff
@@ -0,0 +1,178 @@
+    {
+        return [
+            new FrameworkBundle(),
+        ];
+    }
+
+    protected function configureContainer(ContainerConfigurator $container): void
+    {
+        $container
+            ->services()
+                ->load('Glpi\Controller\\', $this->getProjectDir() . '/src/Glpi/Controller')
+                    ->autowire()
+                    ->autoconfigure()
+        ;
+    }
+
+    protected function configureRoutes(RoutingConfigurator $routes): void
+    {
+        $routes->import($this->getProjectDir() . '/src/Glpi/Controller', 'attribute');
+    }
+
+    public function handle(Request $request, int $type = HttpKernelInterface::MAIN_REQUEST, bool $catch = true): Response
+    {
+        $this->runLegacyRouter($request);
+
+        return parent::handle($request, $type, $catch); // TODO: Change the autogenerated stub
+    }
+
+    private function runLegacyRouter(Request $request): void
+    {
+        if (defined('GLPI_ROOT')) {
+            // This var is defined in {root_dir}/index.php , the legacy front controller
+            return;
+        }
+
+        /**
+         * GLPI web router.
+         *
+         * This router is used to be able to expose only the `/public` directory on the webserver.
+         */
+
+        $glpi_root = realpath(dirname(__DIR__, 3));
+
+        if (
+            $request->server->get('SCRIPT_NAME') === '/public/index.php'
+            && preg_match('/^\/public/', $request->server->get('REQUEST_URI')) !== 1
+        ) {
+            // When requested URI does not start with '/public' but `$request->server->get('SCRIPT_NAME')` is '/public/index.php',
+            // it means that document root is the GLPI root directory, but a rewrite rule redirects the request to the PHP router.
+            // This case happen when redirection to PHP router is made by an `.htaccess` file placed in the GLPI root directory,
+            // and has to be handled to support shared hosting where it is not possible to change the web server root directory.
+            $uri_prefix = '';
+        } else {
+            // `$request->server->get('SCRIPT_NAME')` corresponds to the script path relative to server document root.
+            // -> if server document root is `/public`, then `$request->server->get('SCRIPT_NAME')` will be equal to `/index.php`
+            // -> if script is located into a `/glpi-alias` alias directory, then `$request->server->get('SCRIPT_NAME')` will be equal to `/glpi-alias/index.php`
+            $uri_prefix = rtrim(str_replace('\\', '/', dirname($request->server->get('SCRIPT_NAME'))), '/');
+        }
+
+        // Get URI path relative to GLPI (i.e. without alias directory prefix).
+        $path = preg_replace(
+            '/^' . preg_quote($uri_prefix, '/') . '/',
+            '',
+            parse_url($request->server->get('REQUEST_URI') ?? '/', PHP_URL_PATH)
+        );
+
+        require_once $glpi_root . '/src/Glpi/Http/ProxyRouter.php';
```

Comment:
I think this line is useless. Classes in the `Glpi` namespace are loaded using the native composer PSR4 loader.
```suggestion

        require_once $glpi_root . '/src/Glpi/Http/ProxyRouter.php';
```